java/drivers/driver-hazelcast4plus/conf/worker.sh

```diff
@@ -7,20 +7,112 @@
 # See the end of this file for examples for different profilers.
 #
 
-# automatic exit on script failure
-set -e
+# Enable strict error handling
+set -euo pipefail
+
 # printing the command being executed (useful for debugging)
 #set -x
 
-# redirecting output/error to the right log files
-exec > worker.out
-exec 2> worker.err
+# Redirecting output and error to log files with timestamps
+exec > >(tee -a worker.out) 2> >(tee -a worker.err >&2)
+
+log() {
+    local level="$1"
+    shift
+    echo "$(date '+%Y-%m-%d %H:%M:%S') [$level] $*"
+}
 
 # Read the parameters file and add it to the environment
 while IFS='=' read -r key value; do
     export "$key"="$value"
 done < "parameters"
 
+validate_mount_params() {
+    if [[ -n "${mount_path:-}" && -z "${mount_volume:-}" ]]; then
+        log "ERROR" "mount_path is set but mount_volume is not set."
+        exit 1
+    fi
+
+    if [[ -z "${mount_path:-}" && -n "${mount_volume:-}" ]]; then
+        log "ERROR" "mount_volume is set but mount_path is not set."
+        exit 1
+    fi
+}
+
+validate_mount_path_safety() {
+    local path="$1"
+    case "$path" in
+        / | /home | /var | /etc)
+            log "WARNING" "mount_path $path is not allowed. Aborting."
+            exit 1
+            ;;
+    esac
+}
+
+mount_persistence_volume() {
```

Comment:
Any concerns about the script potentially failing during this, leaving a partially mounted volume etc? Might be a good idea to setup a cleanup function with traps

Reply:
Unlikely to happen but worth doing nonetheless -- see 1b58cae